From df061ee8dcd190d562d5a0d769caa4a86b829d29 Mon Sep 17 00:00:00 2001
From: Ioan Sucan <isucan@gmail.com>
Date: Sat, 15 Feb 2014 19:40:25 -0800
Subject: [PATCH] add install rule for package.xml

---
 CMakeLists.txt | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index fec8048..919a089 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -34,4 +34,7 @@ if (NOT MSVC)
   install(FILES ${pkg_conf_file} DESTINATION lib/pkgconfig/ COMPONENT pkgconfig)
 endif()
 
+# Install catkin package.xml
+install(FILES package.xml DESTINATION share/urdfdom_headers)
+
 message(STATUS "Configuration successful. Type make install to install urdfdom_headers.")
-- 
1.9.0.rc1.175.g0b1dcb5

