From 831191bd10891fdb26bc7a195bc29785ae9bf9ab Mon Sep 17 00:00:00 2001
From: William Woodall <wjwwood@gmail.com>
Date: Tue, 12 Mar 2013 16:35:56 -0700
Subject: [PATCH] Install rule for package.xml

---
 CMakeLists.txt | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 261b9a2..00e2d17 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -34,4 +34,7 @@ if (NOT MSVC)
   install(FILES ${pkg_conf_file} DESTINATION lib/pkgconfig/ COMPONENT pkgconfig)
 endif()
 
+# Install catkin package.xml
+install(FILES package.xml DESTINATION share/urdfdom_headers)
+
 message(STATUS "Configuration successful. Type make install to install urdfdom_headers.")
-- 
1.7.12.4 (Apple Git-37)

