From d53c1bcef5facb80e5e16ea36f26382cd24cc235 Mon Sep 17 00:00:00 2001
From: Ioan Sucan <isucan@willowgarage.com>
Date: Wed, 19 Jun 2013 01:28:19 +0300
Subject: [PATCH 2/2] install as catkin package

---
 CMakeLists.txt |   29 +++++++++++++++++++++--------
 1 file changed, 21 insertions(+), 8 deletions(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index c090c78..1f3c4bf 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -4,6 +4,22 @@ project (urdfdom_headers)
 set (URDF_MAJOR_VERSION 0)
 set (URDF_MINOR_VERSION 2)
 set (URDF_PATCH_VERSION 3)
+set(${PROJECT_NAME}_inc_dirs urdf_exception/include urdf_model/include urdf_model_state/include urdf_sensor/include urdf_world/include)
+find_package(catkin REQUIRED)
+catkin_package(
+  INCLUDE_DIRS ${${PROJECT_NAME}_inc_dirs}
+)
+include_directories(${${PROJECT_NAME}_inc_dirs})
+
+# split version in parts and pass to preprocessor
+string(REPLACE "." ";" ${PROJECT_NAME}_VERSION_LIST "${${PROJECT_NAME}_VERSION}")
+list(LENGTH ${PROJECT_NAME}_VERSION_LIST _count)
+if(NOT _count EQUAL 3)
+  message(FATAL_ERROR "${PROJECT_NAME} version '${${PROJECT_NAME}_VERSION}' does not match 'MAJOR.MINOR.PATCH' pattern")
+endif()
+list(GET ${PROJECT_NAME}_VERSION_LIST 0 URDF_MAJOR_VERSION)
+list(GET ${PROJECT_NAME}_VERSION_LIST 1 URDF_MINOR_VERSION)
+list(GET ${PROJECT_NAME}_VERSION_LIST 2 URDF_PATCH_VERSION)
 
 set (URDF_VERSION ${URDF_MAJOR_VERSION}.${URDF_MINOR_VERSION}.${URDF_PATCH_VERSION})
 
@@ -23,18 +39,15 @@ add_subdirectory(urdf_exception)
 
 set(PACKAGE_NAME ${PROJECT_NAME})
 set(cmake_conf_file "${CMAKE_CURRENT_SOURCE_DIR}/cmake/${PROJECT_NAME}-config.cmake")
-configure_file("${cmake_conf_file}.in" "${cmake_conf_file}" @ONLY)
-install(FILES ${cmake_conf_file} DESTINATION share/${PROJECT_NAME}/cmake/ COMPONENT cmake)
+#configure_file("${cmake_conf_file}.in" "${cmake_conf_file}" @ONLY)
+#install(FILES ${cmake_conf_file} DESTINATION share/${PROJECT_NAME}/cmake/ COMPONENT cmake)
 
 # Make the package config file
 if (NOT MSVC)
   set(PACKAGE_DESC "Unified Robot Description Format")
   set(pkg_conf_file "${CMAKE_CURRENT_SOURCE_DIR}/cmake/pkgconfig/urdfdom_headers.pc")
-  configure_file("${pkg_conf_file}.in" "${pkg_conf_file}" @ONLY)
-  install(FILES ${pkg_conf_file} DESTINATION lib/pkgconfig/ COMPONENT pkgconfig)
+  #configure_file("${pkg_conf_file}.in" "${pkg_conf_file}" @ONLY)
+  #install(FILES ${pkg_conf_file} DESTINATION lib/pkgconfig/ COMPONENT pkgconfig)
 endif()
 
-# Install catkin package.xml
-install(FILES package.xml DESTINATION share/urdfdom_headers)
-
-message(STATUS "Configuration successful. Type make install to install urdfdom_headers.")
+#message(STATUS "Configuration successful. Type make install to install urdfdom_headers.")
-- 
1.7.9.5

